src/Backend/Modules/MediaGalleries/Domain/MediaGallery/MediaGalleryRepository.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+namespace Backend\Modules\MediaGalleries\Domain\MediaGallery;
+
+use Doctrine\ORM\EntityRepository;
+use Backend\Modules\MediaGalleries\Domain\MediaGallery\Exception\MediaGalleryNotFound;
+
+/**
+ * MediaGallery Repository
+ */
+final class MediaGalleryRepository extends EntityRepository
+{
+    /**
+     * Add a MediaGallery
+     *
+     * @param MediaGallery $mediaGallery
+     *
+     * We don't flush here, see http://disq.us/p/okjc6b
+     */
+    public function add(MediaGallery $mediaGallery)
+    {
+        $this->getEntityManager()->persist($mediaGallery);
+    }
+
+    /**
+     * Exists MediaGallery by title
+     *
+     * @param string $title
+     * @param int $ignoreMediaGalleryId
+     * @return boolean
+     */
+    public function existsByTitle(string $title, int $ignoreMediaGalleryId = 0): bool
+    {
+        /** @var MediaGallery $mediaGallery */
+        $mediaGallery = $this->findOneByTitle($title);
+
+        return ($mediaGallery)
+            ? ($mediaGallery->getId() != $ignoreMediaGalleryId) : false;
+    }
+
+    /**
+     * @param string|null $id
+     * @return MediaGallery
+     * @throws \Exception
+     */
+    public function getOneById(string $id): MediaGallery
+    {
+        if ($id === null) {
```

Comment:
the typehint doesn't allow null to happen here, if you want to allow it, like stated in the dock blocks you need to add `= null`